sql/opt_range.cc

```diff
@@ -5475,10 +5547,11 @@ TABLE_READ_PLAN *get_best_disjunct_quick(PARAM *param, SEL_IMERGE *imerge,
   DBUG_PRINT("info", ("index_merge scans cost %g", imerge_cost));
   trace_best_disjunct.add("cost_of_reading_ranges", imerge_cost);
 
-  if (imerge_too_expensive || (imerge_cost > read_time) ||
-      ((non_cpk_scan_records+cpk_scan_records >=
-        param->table->stat_records()) &&
-       read_time != DBL_MAX))
+  if (imerge_too_expensive || (((imerge_cost > read_time) ||
+                               ((non_cpk_scan_records+cpk_scan_records >=
+                                 param->table->stat_records()) &&
+                                read_time != DBL_MAX)) &&
+                               !force_index_merge))
```

Comment:
Do I understand it correctly:  if `imerge_too_expensive==true`  then we will take this if-branch, execute this code

```cpp
    trace_best_disjunct.add("chosen", false).add("cause", "cost");
    DBUG_RETURN(NULL);
```

ignoring the fact that `force_index_merge=true` due to hints? 
Is this intended behavior? 

Also, I think the condition in the if statement became unreadable and needs to be written in a more human-readable way.